substrate/frame/contracts/src/lib.rs

```diff
@@ -627,7 +547,7 @@ pub mod pallet {
 		/// To avoid this situation a constructor could employ access control so that it can
 		/// only be instantiated by permissioned entities. The same is true when uploading
 		/// through [`Self::instantiate_with_code`].
-		#[pallet::call_index(3)]
+		#[pallet::call_index(0)]
```

Comment:
you should never change index

Reply:
yea please revert any changes to the contracts logic and just copy&paste the `OldWeight` type into it.

Reply:
yep, that's already done. Can you run the `bot fmt`? I'm unable to run it because I'm not a member.

Reply:
actually, let me just close this one and create a new PR, this might be easier.

Reply:
call index determines the encoded value of the extrinsic. change index means a tx generated with old runtime will be decoded to a different extrinsic and that’s a security risk. that’s why we always explicitly annotate the call index and should never change them to keep compatibility. 
you also deleted some extrinsic and that means they won’t be available anymore. and again that’s a breaking change which requires bit more process (eg announce it to ensure ecosystem tools are all updated)
so to keep the scope of this particular PR minimal and safe you from doing extra process work, best to keep the pallet contract intact

Reply:
thank you, that makes sense. Appreciate you taking the time to share your valuable knowledge